libcontainer/rootfs_linux.go

```diff
@@ -73,6 +74,10 @@ func prepareRootfs(pipe io.ReadWriter, iConfig *initConfig, mountFds mountFds) (
 		return fmt.Errorf("malformed mountFds slice. Expected size: %v, got: %v. Slice: %v", len(config.Mounts), len(mountFds.sourceFds), mountFds.sourceFds)
 	}
 
+	if mountFds.idmapFds != nil && len(mountFds.idmapFds) != len(config.Mounts) {
+		return fmt.Errorf("malformed idmapFds slice. Expected size: %v, got: %v. Slice: %v", len(config.Mounts), len(mountFds.idmapFds), mountFds.idmapFds)
```

Comment:
I see the existing error also does so, but I'm on the fence if printing `Slice: %v` is very useful, because the length of the size is the error, and the actual values in it not so much (?). The error message will just print a slice of int's (not sure if that helps debugging issues?).

As this is net-new code, perhaps also fix the "errors should not be capitalised of punctuation" to keep linters more happy; perhaps something like;

```go
return fmt.Errorf("malformed idmapFds slice: expected size: %v, got: %v", len(config.Mounts), len(mountFds.idmapFds))
```